From 801745d5e8cd932fde0b6083d645b5e2dcffefc0 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 10 Oct 2013 15:21:22 +0800
Subject: [PATCH 48/56] libmediaplayerservice: postpone seek action if decoders
 are not established

BZ: 141647

postpone seek action if decoders are not established, or else if client only
parse one track TS packet when seek happened, discontinuity will queue to only one track,
which will cause the audio, video track mismatch and anr happens

Change-Id: I0f196a6efacf49d765f3a8e5b89ec44d60746bcc
Orig-Change-Id: Ifbb0b1fe8be32ffbb8c75fe5e8d052acffce9dc3
Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Reviewed-on: http://android.intel.com:8080/137463
Reviewed-by: Huang, Yuanjun <yuanjun.huang@intel.com>
Reviewed-by: Shi, PingX <pingx.shi@intel.com>
Tested-by: Shi, PingX <pingx.shi@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libmediaplayerservice/nuplayer/NuPlayer.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index 3e695a6..8de568c 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -840,6 +840,12 @@ void NuPlayer::onMessageReceived(const sp<AMessage> &msg) {
 
         case kWhatSeek:
         {
+            if (mAudioDecoder == NULL && mVideoDecoder == NULL) {
+                ALOGV("postpone seek since decoders are not established");
+                msg->post(100000ll);
+                break;
+            }
+
             int64_t seekTimeUs;
             CHECK(msg->findInt64("seekTimeUs", &seekTimeUs));
 
-- 
2.7.4

